sale_commission_product_criteria/models/commission.py

```diff
@@ -0,0 +1,299 @@
+# © 2023 ooops404
+# License AGPL-3 - See https://www.gnu.org/licenses/agpl-3.0.html
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.tools import float_repr
+
+COMMISSION_ITEM_TYPES = {
+    "0_product_variant": 1000,
+    "1_product": 10000,
+    "2_product_category": 100000,
+    "3_global": 1000000,
+}
+
+
+class SaleCommission(models.Model):
+    _inherit = "sale.commission"
+
+    commission_type = fields.Selection(
+        selection_add=[("product", "Product criteria")],
+        ondelete={"product": "set default"},
+    )
+    item_ids = fields.One2many("commission.item", "commission_id", copy=True)
+    sequence = fields.Integer(
+        "Sequence", default=1, help="The first in the sequence is the default one."
+    )
+
+    def write(self, vals):
+        res = super(SaleCommission, self).write(vals)
+        self.sort_items()
```

Comment:
Instead of this, order directly by the selection field.

Reply:
This method will be inherited in further modules:
https://github.com/OCA/commission/pull/421/files#diff-db23ccfdfe6db97ef62e546b8909cb7e32eeb9e56e8f3087d0670decc07f83a2R36

Reply:
You can do the same adding a value on the selection field. Doing this is counter-productive in performance terms. If you have to put values in between, name the selection options `10_`, `20_` instead of `1_`, `2_`, and put the values in between like `15_` or similar.

Reply:
@pedrobaeza Perhaps this code should be implemented on commission.item level to improve performance a bit. But i dont understand how selection can help here. Quantity of commission items can be unpredicatable big. Why is it wrong to use sequence for sorting, which is standard odoo approach?

Reply:
The problem is to sort on each write/create. This can also create loops very difficult to debug. You can sort by any needed field in the model. If quantity is a criteria, then sort by it.

Reply:
ok. it has been redone to use only **_order**.